gping.py

```diff
@@ -212,11 +239,45 @@ def __receive__(self):
                 # delete the ping
                 del(self.pings[packet_id])
 
+    def print_failures(self):
+        print >>sys.stderr
+        print >>sys.stderr, 'Failures:'
+        template = '{hostname:45}{message}'
+        for failure in self.failures:
+            message = template.format(hostname=failure['dest_addr'], message='message' in failure and failure['message'])
```

Comment:
It would be nicer to have a default message instead of `False`.

```python
message = template.format(hostname=failure['dest_addr'], message=failure.get('message', 'unknown error'))
```